libraries/resource_providers/api200/server_profile_provider.rb

```diff
@@ -30,6 +30,30 @@ def set_connections(type, connection_list)
         true
       end
 
+      def build_volume_attachments
+        return unless @new_resource.volume_attachments
```

Comment:
This clause is useless since you are defaulting the volume_attachments to `[]`, therefore it would never return here.

Reply:
Even if user set the value to nil?

Reply:
really it is not necessary, if user is setting the value to nil, he is using the recipe in wrong way.

Reply:
By one side the user really wants to set it to nil regardless of the recommendations to leave it alone when he does not wants any attachment, becoming susceptible to unexpected failures. But, by other side, the user may inadvertedly put it that way.

Feel free to change it or not.